addons/rating/controllers/main.py

```diff
@@ -26,6 +26,13 @@ def action_open_rating(self, token, rate, **kwargs):
 
         rating, record_sudo = self._get_rating_and_record(token)
 
+        if not request.env.user._is_public() and \
+                request.env.user.partner_id.commercial_partner_id != rating.partner_id.commercial_partner_id:
+            return request.render('rating.rating_external_page_invalid_partner', {
+                'model_name': request.env['ir.model']._get(rating.res_model).display_name,
+                'name': record_sudo.name,
```

Comment:
Better use 'display_name' for that purpose, as it correctly checks for rec_name
```suggestion
                'name': record_sudo.display_name,
```